Cabal/Cabal.cabal

```diff
@@ -315,6 +315,7 @@ library
     Distribution.Compat.Exception
     Distribution.Compat.Graph
     Distribution.Compat.Internal.TempFile
+    Distribution.Compat.Prelude
```

Comment:
I actually started the refactoring using `base-compat`, but then realised we'd end up with an additional dependency to keep track of in bootstrap.sh as well as redundant/duplicate maintenance work keeping `Distribution.Compat.Client.Prelude` and `Distribution.Compat.Prelude` in sync in order to have a common vocabulary between `Cabal` and `cabal-install` which are essentially the same project spread across two packages.